packages/frontend/src/components/UserSettings/AccessTokensPanel/TokensTable.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, Classes, Dialog } from '@blueprintjs/core';
+import { ApiPersonalAccessTokenResponse, formatDate } from '@lightdash/common';
+import { Button as MantineButton, Paper, Table } from '@mantine/core';
+import { FC, useState } from 'react';
+import { useTableStyles } from '../../../hooks/styles/useTableStyles';
+import {
+    useAccessToken,
+    useDeleteAccessToken,
+} from '../../../hooks/useAccessToken';
+
+const TokenItem: FC<{
+    token: ApiPersonalAccessTokenResponse;
+}> = ({ token }) => {
+    const { description, expiresAt, uuid } = token;
+    const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+    const { mutate, isLoading: isDeleting } = useDeleteAccessToken();
+
+    return (
+        <>
+            <tr>
+                <td>{description}</td>
+                <td>
+                    {expiresAt ? formatDate(expiresAt) : 'No expiration date'}
+                </td>
+                <td>
+                    <MantineButton
+                        variant="outline"
+                        size="xs"
+                        color="red"
+                        onClick={() => setIsDeleteDialogOpen(true)}
+                    >
+                        Delete
+                    </MantineButton>
+                </td>
+            </tr>
+            <Dialog
```

Comment:
is it possible to remove modal rendering for every single item or can it be a 1 modal for all access tokens but open with a proper data inside a modal?

Reply:
Thanks, just did it! 👨‍🍳